JinGine.Domain/JinGine.Domain.csproj

```diff
@@ -6,4 +6,8 @@
 	<LangVersion>10</LangVersion>
   </PropertyGroup>
 
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Extensions.Primitives" Version="7.0.0" />
+  </ItemGroup>
+
```

Comment:
if it ends up being easier to work with a char array than a string, just don't use this package and use ArraySegment objects where StringSegment objects were used.